eslint.config.mjs

```diff
@@ -0,0 +1,43 @@
+// @ts-check
+
+import eslint from '@eslint/js'
+import prettierRecommended from 'eslint-plugin-prettier/recommended'
+import tseslint from 'typescript-eslint'
+import globals from 'globals'
+
+export default tseslint.config(
+  {
+    ignores: ['dist', 'jest.config.ts'],
+  },
+  eslint.configs.recommended,
+  ...tseslint.configs.strictTypeChecked,
+  prettierRecommended,
+  {
+    languageOptions: {
+      globals: globals.node,
+      parserOptions: {
+        project: './tsconfig.json',
+        tsconfigDirName: import.meta.dirname,
+      },
+    },
+    linterOptions: {
+      reportUnusedDisableDirectives: 'warn',
+    },
+    rules: {
+      'max-lines': ['warn', 500],
```

Comment:
I ended up bumping this to `2000` after the tsdocs where introduced in #384. Should we just disable this rule entirely?

 ```suggestion
      'max-lines': ['warn', 2000],
```